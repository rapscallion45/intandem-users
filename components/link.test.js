import React from 'react';
import { shallow } from 'enzyme';
import router from 'next/router';
import toJson from 'enzyme-to-json';
import Link from './link';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
}));

describe('Link', () => {
  it('Renders correctly enzyme', () => {
    /* Arrange */
    const useRouter = jest.spyOn(router, 'useRouter');
    useRouter.mockImplementation(() => ({
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    }));

    /* Act */
    const wrapper = shallow(<Link href="/" />);

    /* Assert */
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Options', () => {
    it('renders external link correctly', () => {
      /* Arrange */
      const useRouter = jest.spyOn(router, 'useRouter');
      useRouter.mockImplementation(() => ({
        route: '/',
        pathname: '',
        query: '',
        asPath: '',
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
      }));

      /* Act */
      const wrapper = shallow(<Link href="http://test.com" />);

      /* Assert */
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders external link with no style correctly', () => {
      /* Arrange */
      const useRouter = jest.spyOn(router, 'useRouter');
      useRouter.mockImplementation(() => ({
        route: '/',
        pathname: '',
        query: '',
        asPath: '',
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
      }));

      /* Act */
      const wrapper = shallow(<Link href="http://test.com" noLinkStyle />);

      /* Assert */
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders link with no style correctly', () => {
      /* Arrange */
      const useRouter = jest.spyOn(router, 'useRouter');
      useRouter.mockImplementation(() => ({
        route: '/',
        pathname: '',
        query: '',
        asPath: '',
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
      }));

      /* Act */
      const wrapper = shallow(<Link href="/" noLinkStyle />);

      /* Assert */
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
