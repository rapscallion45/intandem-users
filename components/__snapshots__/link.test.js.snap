// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Options renders external link correctly 1`] = `
<ForwardRef(Link)
  className=""
  href="http://test.com"
/>
`;

exports[`Link Options renders external link with no style correctly 1`] = `
<Styled(a)
  className=""
  href="http://test.com"
/>
`;

exports[`Link Options renders link with no style correctly 1`] = `
<ForwardRef
  className=""
  to="/"
/>
`;

exports[`Link Renders correctly enzyme 1`] = `
<ForwardRef(Link)
  className=""
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "href": [Function],
        "linkAs": [Function],
        "locale": [Function],
        "passHref": [Function],
        "prefetch": [Function],
        "replace": [Function],
        "scroll": [Function],
        "shallow": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  to="/"
/>
`;
